plugins/scaffolder/src/components/PrefilledWorkflowLink/usePrefilledWorkflowHref.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useRouteRef } from '@backstage/core-plugin-api';
+import { selectedTemplateRouteRef } from '../../routes';
+import qs from 'qs';
+import type { PrefilledWorkflowProps } from './types';
+
+/**
+ * Hook that creates prefilled workflow url that can be used programmatically in something like navigate
+ *
+ * @public
+ */
+export function usePrefilledWorkflowHref({
+  templateName,
+  namespace,
+  formData,
+}: PrefilledWorkflowProps): string {
+  const templateRoute = useRouteRef(selectedTemplateRouteRef);
```

Comment:
We can't export the route ref here, not even indirectly like this. RouteRefs are internal and need to be linked together with other sources through ExternalRouteRefs. There's one exception to this, which is the entity page route ref over [here](https://github.com/backstage/backstage/blob/dd6fa27a9f1d0f6335585c2a00e83cc3e21fcee6/plugins/catalog-react/src/routes.ts#L31), but as you can see there's some extra stuff needed to make that work, and it needs to be a very stable route ref.

I'd say that in the end the current routing system doesn't support what you're aiming for here, and I think we need to step back and figure out how one would do this in a clean way, extending the framework as necessary. I'm thinking that for now we might settle for documenting this pattern? It's not a lot of code to be repeating.

Reply:
@Rugvip thank you for the reply.  What is a `stable route ref` in this context?

Would the goal still be to try and get a reference to the selected template, something like:

```ts
const templateRouteRef = getOrCreateGlobalSingleton(
  'catalog:template-route-ref',
  () =>
    createSubRouteRef({
      id: 'scaffolder/selected-template',
      parent: ??,
      path: '/templates/:namespace/:templateName',
    }),
);
```